src/checkin/services/flow/step/GetExtraPassengerInformationStep.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { MockContext } from '@testMocks/model/Context.mock'
+import { Context } from '../../../model/Context'
+import GetExtraPassengerInformationStep from './GetExtraPassengerInformationStep'
+
+const MOCK_NAMES = ['Carlos', 'Juan', 'Emilio', 'Gary', 'Elias', 'Fernando', 'Giovanni']
+const MOCK_PASSPORTS = ['G143327', 'G113125', 'G229108', 'G229101']
+
+describe('[ Step / GetExtraPassengerInformationStep ]', () => {
+  const step = new GetExtraPassengerInformationStep()
+
+  describe('Test when it should be execute', () => {
+    it('should return true when passengers are more than one', () => {
+
+      const context = wireContextMockWithNumberOfPassengersInSessionData(2)
+
+      expect(step.when(context)).toBeTruthy()
+    })
+
+    it('should return false when there is only one passenger', () => {
+      const context = wireContextMockWithNumberOfPassengersInSessionData(1)
+
+      expect(step.when(context)).toBeFalsy()
+    })
+  })
+
+  describe('Test at execution', () => {
+    it('should request the extra passengers information', async () => {
+      const context: Context = wireContextMockWithNumberOfPassengersInSessionData(2)
+      wireContextMockWithNullExtraPassengerInformationInRequest(context)
+
+      const response = await step.onExecute(context)
+
+      expect(response).toBeFalsy()
+      expect(context.getResponse().requiredFiles?.extra_passenger_information).toBeNull()
+      expect(context.getSession().extra_passenger_information).toBeUndefined()
+    })
+
+    it('should request the exact amount of extra passengers information', async () => {
+      const context = wireContextMockWithNumberOfPassengersInSessionData(2)
+      wireContextMockWithAmountOfExtraPassengersInformationInRequest(0, context)
+
+      const response = await step.onExecute(context)
+
+      expect(response).toBeFalsy()
+      expect(context.getResponse().requiredFiles?.extra_passenger_information).toBeNull()
+      expect(context.getSession().extra_passenger_information).toBeUndefined()
+    })
+
+    it('should set the extra passenger information in session if it is given by the user', async () => {
+      const context = wireContextMockWithNumberOfPassengersInSessionData(2)
+      wireContextMockWithAmountOfExtraPassengersInformationInRequest(1, context)
+      const mockExtraPassengerInformation = context.getRequest().fields?.extra_passenger_information as Array<{ name: string, passport: string }>
+
+      const response = await step.onExecute(context)
+
+      expect(response).toBeTruthy()
+      expect(context.getSession().extra_passenger_information).toBe(mockExtraPassengerInformation)
+    })
+  })
+})
+
+function wireContextMockWithNumberOfPassengersInSessionData(numberOfPassengers: number, context = new MockContext()): Context {
+  context.withSessionBuilder(sessionBuilder => sessionBuilder
+    .data({
```

Comment:
Instead of explicitly listing and assigning all the fields of the data property, you could destructure the existing data in `context.getSession()` and copy them to the new object without needing to assign them one by one, only adding the new field, in this case, passangers.  For example:
`.data({ ...context.getSession().data, passengers: numberOfPassengers})`

Reply:
Thanks for the suggestion